OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo

```diff
@@ -192,6 +194,82 @@ algorithm
   end if;
 end removeVerySimpleEquations;
 
+public function fixAliasVarsVariablity
+" This is a workaround for #5673
+  This module traverses all alias variables and double-checks if the referenced alias variables variablitly
+  is parameter or not and changes variablilty of alias variable to BackendDAE.PARAM()."
+  input BackendDAE.BackendDAE inDAE;
+  output BackendDAE.BackendDAE outDAE;
+protected
+  BackendDAE.Variables aliasVars,systvars,globalKnownVars;
+  DAE.Exp binding;
+  DAE.ComponentRef cr;
+  Boolean param,found;
+  BackendDAE.Shared shared;
+  list<BackendDAE.Var> referencevar,knownVarList={},aliasVarList={};
+  BackendDAE.Var tempvar;
+algorithm
+  aliasVars := BackendDAEUtil.getAliasVars(inDAE);
+  systvars := getVarsFromEqSystem(inDAE.eqs);
+  //BackendDump.dumpVariables(aliasVars,"aliasVariable-Actual");
+  for var in BackendVariable.varList(aliasVars) loop
+    binding := BackendVariable.varBindExp(var);
+    {cr} := Expression.getAllCrefs(binding);
+    // look up for the reference variable variablity
+    (referencevar,found) := getVarsHelper(cr,systvars); // first check in EqSystem.OrderedVars
+    //check in shared.globalKnownVars if not found in orderedVars
+    if(not found) then
+      (referencevar,found) := getVarsHelper(cr,inDAE.shared.globalKnownVars);
+    end if;
+    param := List.mapAllValueBool(referencevar, BackendVariable.isParam, true);
```

Comment:
This doesn't consider constants, e.g. const + param gives false but should be true.